gym/lib/gym/runner.rb

```diff
@@ -18,6 +18,9 @@ def run
         build_app
       end
       verify_archive unless Gym.config[:skip_archive]
+
+      return nil if Gym.config[:skip_archive]
```

Comment:
FWIW, this single line here seemed to break our use of
```ruby
build_ios_app(skip_build_archive: true, skip_archive: true, export_options: {method: "ad-hoc"}, ...)
```
for exporting a freshly built and archived build under a different provisioning mode (e.g. one to be uploaded to TestFlight and another as Ad Hoc to an internal distribution system).

I don't know if it was ever intended of `gym` to support that use case, so maybe it's not a bug but I should request (or contribute) it as a new action instead. (Curiously, the existing `resign` action doesn't seem to support our use case AFAICT.)

For now, we can just use `xcodebuild -exportArchive -archivePath <xcarchive> -exportOptionsPlist <plist> -exportPath <dir>` with `sh`, but would be nice if it were supported better by Fastlane out of the box.